src/Sulu/Bundle/AdminBundle/Resources/js/components/RectangleSelection/tests/RectangleSelection.test.js

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+import React from 'react';
+import {render, mount} from 'enzyme';
+import RectangleSelection from '../RectangleSelection';
+
+class MockedRectangleSelection extends RectangleSelection {
+    constructor(props) {
+        super(props);
+        this.setContainer = () => this.container = {clientWidth: 2000, clientHeight: 1000};
+        this.spyed = false;
+        this.componentDidUpdate = () => {
+            if (!this.spyed) {
+                this.spyed = true;
+                props.updateSpy();
+            }
+        };
+    }
+}
+
+test('The component should render with children', () => {
+    const view = render(
+        <RectangleSelection>
+            <p>Lorem ipsum</p>
+        </RectangleSelection>
+    );
+
+    expect(view).toMatchSnapshot();
+});
+
+test('The component should render with initial selection', (done) => {
+    window.requestAnimationFrame = jest.fn((cb) => cb());
+
+    let view;
+    const spy = () => {
+        expect(view.render()).toMatchSnapshot();
+        done();
+    };
+
+    view = mount(
+        <MockedRectangleSelection updateSpy={spy} initialSelection={{width: 1, height: 2, top: 3, left: 4}}>
+            <p>Lorem ipsum</p>
+        </MockedRectangleSelection>
+    );
+});
+
+test('The component should maximize the selection when no initial values given', (done) => {
+    window.requestAnimationFrame = jest.fn((cb) => cb());
+
+    let view;
+    const spy = () => {
+        expect(view.render()).toMatchSnapshot();
+        done();
+    };
+
+    view = mount(
+        <MockedRectangleSelection updateSpy={spy}>
+            <p>Lorem ipsum</p>
+        </MockedRectangleSelection>
+    );
+});
+
+test('The component should center and maximize the selection when a minHeight and minWidth is given', (done) => {
+    window.requestAnimationFrame = jest.fn((cb) => cb());
+
+    let view;
+    const spy = () => {
+        expect(view.render()).toMatchSnapshot();
+        done();
+    };
+
+    view = mount(
+        <MockedRectangleSelection updateSpy={spy} minHeight={200} minWidth={50}>
+            <p>Lorem ipsum</p>
+        </MockedRectangleSelection>
+    );
+    view = mount(
+        <MockedRectangleSelection updateSpy={spy} minHeight={50} minWidth={200}>
+            <p>Lorem ipsum</p>
+        </MockedRectangleSelection>
+    );
+});
+
+test('The component should publish the new selection when the rectangle changes', (done) => {
+    window.requestAnimationFrame = jest.fn((cb) => cb());
+
+    let view;
+    let selection = {};
+    const setSelection = s => selection = s;
```

Comment:
I think we should enforce `const setSelection = (s) => selection = s;`, because it is more readable IMO.